doc/tutorials/outsourced-cluster.rst

```diff
@@ -0,0 +1,106 @@
+##################################
+Outsourced Distributed Computation
+##################################
+
+Secure XGBoost is tailored toward an outsourced computation model -- one in which a client with sensitive data wants to outsource computation on the sensitive data to an untrusted server with a trusted hardware enclave. With a lot of training data, a cluster of machines, as opposed to a single server, may have to be used. This tutorial provides an example of such a scenario. 
+
+**Contents**
+
+* `Architecture`_
+
+* `Tutorial`_
+
+  - `Cluster Setup`_
+  - `Client Setup`_
+
+* `Troubleshooting`_
+
+************
+Architecture
+************
+In this scenario there's a client and an enclave cluster. The client communicates with each machine in the cluster over RPC, and the nodes in the cluster communicate with one another over TLS channels that begin and end inside an enclave.
+
+.. image:: images/outsourced-distributed.png
+   :width: 100% 
+   :alt: Outsourced Distributed Architecture
+
+Before computation, one node in the cluster will initialize a tracker to start an RPC server on each node in the cluster. The tracker is discussed in the :doc:`Distributed Secure XGBoost tutorial <distributed>`.
+
+Each RPC server will listen for client requests. The client will make requests for remote attestation, adding its symmetric key used to encrypt the training data to each enclave, and the end-to-end training process. 
+
+The client will first attest each enclave, then encrypt the data and send it to the cluster. The data remains encrypted until an enclave at each machine loads a part of the data, at which point the enclave can decrypt the data using the symmetric key sent by the client over RPC. Once the client has given the start signal to each node over RPC, the cluster will then collectively train a model on the training data following the distributed training pattern outlined in the :doc:`Distributed Secure XGBoost tutorial <distributed>`. Lastly, an encrypted version of the trained model will be saved, available for the client to fetch. 
```

Comment:
I think we should change the future tense to present tense (in this para + the previous one)?

Also, rephrasing the 2nd sentence to the following reads better: "Each enclave in the cluster loads a subset of the encrypted data, and decrypts it using symmetric key sent by the client over RPC."